markdown/events/2021/bytesize-5-dsl2-module-development.md

```diff
@@ -32,3 +32,257 @@ This week, Harshil Patel ([@drpatelh](http://github.com/drpatelh/)) will present
+
+So, every time this process is run in the terminal it will print the sample name and also whether the sample is single end or not and now this offers another layer flexibility where the module doesn't have to be or the pipeline itself doesn't have to deal with the single endedness or the paired endedness of the sample, the module is now dealing with that in itself so you can have a mixture of single end and paired end samples being provided to the main workflow and the module will take care of that.
+
+[34:05](https://www.youtube.com/watch?v=ggGGhTMgyHI&t=2045)
+That was the whole philosophy behind that. Meta, again, may not be required in all instances because you may, for example, just be indexing a genome and any parameters that must be evaluated in the context of the sample must be provided within the process so single and paired and, again, to allow that flexibility we need to add those if else statements for whether things are single end or paired within the module file itself.
+
+[34:27](https://www.youtube.com/watch?v=ggGGhTMgyHI&t=2067)
+Hopefully, to deal quite efficiently with file formats and correction and so on, compressed is good. The output if you must emit a name channel. This is useful because for someone else that wants to use your module that you've submitted to nf-core/modules, it helps for them to be able to access the elements that are being produced by this output individually.
+
+[34:54](https://www.youtube.com/watch?v=ggGGhTMgyHI&t=2094)
+You might have 10 or 20 different files being produced by the module and so it helps to have a standard named convention for this.
+
+So, if meta is provided as an input, it must be provided as an output and, again, compressed files are great. The script section, as I mentioned earlier, the software name is automatically obtained by this `task.process` Nextflow variable. A command must be provided to get the software version when you submit this module. This has become really useful when we're collating the software versions at the end of a pipeline.
+
+If this is sort of dealt out and propagated to the module file itself, then we don't have to worry about that. It's done once and you forget about it.
+
+[35:39](https://www.youtube.com/watch?v=ggGGhTMgyHI&t=2139)
+So, only define input and output files as command line parameters, as I mentioned before, so you you don't define optional arguments. These must come in via this optional $options.args. Similarly, anything that requires (or can use) the Nextflow $task.cpus memory or time has to be, actually time maybe not,  but cpus and memory, hopefully in cases where you actually really have to do it because memory gets tricky.
```

Comment:
```suggestion
So, only define input and output files as command line parameters, as I mentioned before, so you you don't define optional arguments. These must come in via this optional $options.args. Similarly, anything that requires (or can use) the Nextflow $task.cpus memory or time has to be, actually time maybe not, but cpus and memory, hopefully in cases where you actually really have to do it because memory gets tricky.
```